File: src/session/play.c
/**
 * play.c — Play flow
 */
#include "session/play.h"
#include "core/log.h"
#include "librtmp2/types.h"

int lrtmp2_play_begin(lrtmp2_conn_t *conn, const char *stream_name)
{
    if (!conn) return LRTMP2_ERR_INTERNAL;
    LRTMP2_LOG_INFO("Begin play: %s", stream_name);
    return LRTMP2_OK;
}

File: src/session/play.rs
//! Play flow
//!
//! Mirrors `src/session/play.h` and `src/session/play.c`.

use super::stream::Stream;
use crate::types::Result;

/// Begin playing on a stream.
pub fn play_begin(stream: &mut Stream, stream_name: &str) -> Result<()> {
    stream.is_playing = true;
    stream.name = stream_name.to_string();
    Ok(())
}
